src/images/images.controller.ts

```diff
@@ -1,7 +1,81 @@
-import { Controller } from '@nestjs/common';
+import {
+  BadRequestException,
+  Controller,
+  Param,
+  ParseIntPipe,
+  Post,
+  UploadedFile,
+  UploadedFiles,
+  UseGuards,
+  UseInterceptors,
+} from '@nestjs/common';
+import { FileFieldsInterceptor } from '@nestjs/platform-express/multer/interceptors/file-fields.interceptor';
+import { FileInterceptor } from '@nestjs/platform-express';
+import { AuthGuard } from '@nestjs/passport';
+import {
+  PostFileUploadSwagger,
+  PostThumbnailUploadSwagger,
+  User,
+} from '../common/decorators';
 import { ImagesService } from './images.service';
+import { ApiBearerAuth, ApiTags } from '@nestjs/swagger';
 
-@Controller('images')
+const imageType = [
+  'image/png',
+  'image/jpg',
+  'image/webp',
+  'image/jpeg',
+  'image/gif',
+];
+
+@ApiBearerAuth('accessToken')
+@ApiTags('image API')
+@Controller('uploads')
 export class ImagesController {
   constructor(private readonly imagesService: ImagesService) {}
+
+  @PostFileUploadSwagger()
+  @Post('/')
+  @UseGuards(AuthGuard('jwt'))
+  @UseInterceptors(
+    FileFieldsInterceptor([{ name: 'files', maxCount: 10 }], {
+      fileFilter: (req, file, callback) => {
+        const ext = file.mimetype;
+        if (!imageType.includes(ext)) {
+          callback(new BadRequestException('이미지 타입만 가능'), false);
+        }
+        callback(null, true);
+      },
+    }),
+  )
+  uploadFile(@UploadedFiles() files: { files: Express.MulterS3.File[] }) {
+    return this.imagesService.uploadFile(files);
+  }
+
+  @PostThumbnailUploadSwagger()
+  @Post('/thumbnail/:boardId')
+  @UseGuards(AuthGuard('jwt'))
+  @UseInterceptors(
+    FileInterceptor('thumbnail', {
+      fileFilter: (req, file, callback) => {
+        const ext = file.mimetype;
+        if (!imageType.includes(ext)) {
+          callback(new BadRequestException('이미지 타입만 가능'), false);
+        }
+        callback(null, true);
+      },
+    }),
+  )
+  uploadThumbnail(
+    @UploadedFile() thumbnail: Express.MulterS3.File,
+    @Param('boardId', ParseIntPipe) boardId: number,
+    @User() userId: number,
+  ) {
+    const dto = {
+      thumbnail,
+      userId,
+      boardId,
+    };
+    return this.imagesService.uploadThumbnail(dto);
```

Comment:
```suggestion
    return this.imagesService.uploadThumbnail({
      thumbnail,
      userId,
      boardId,
    });
```

클린 코드 관련 자료를 공부할 때 인수로 데이터 형식을 그대로 넘겨주는 게 좋다고 하더라고요. 이유는 객체를 선언한 후 인수로 넘겨주면 객체 가공의 유혹을 불러일으킨다네요~
예를들어 dto 객체에 새로운 키 밸류를 넣는다면 uploadThumbnail 메서드 인수로 들어가는 dto의 변화를 예측하기 힘듭니다.
```javascript
    const dto = {
      thumbnail,
      userId,
      boardId,
    };

    // 다양한 코드들...

    dto.name = "새로운 키밸류"

    // 다양한 코드들...

    return this.imagesService.uploadThumbnail(dto);
```

근데 간단한 로직에서는 변화가 크게 없으니 적극적인 고려 사항은 아니니 뭐..😅